dali/operators/reader/loader/video/frames_decoder_gpu.cc

```diff
@@ -529,7 +545,12 @@ bool FramesDecoderGpu::SendFrameToParser() {
 
   // Store pts from current packet to indicate,
   // that this frame is in the decoder
-  piped_pts_.push(av_state_->packet_->pts);
+  if (av_state_->packet_->pts != AV_NOPTS_VALUE) {
+    piped_pts_.push(av_state_->packet_->pts);
+  } else {
+    piped_pts_.push(frame_index_if_no_pts_);
+    frame_index_if_no_pts_++;
+  }
```

Comment:
Is it possible that some frames in the video have PTS and some not?

Reply:
Based on my understanding this should not happen in well formed files. If there is no PTS value, this assumes that the whole video is constant frame rate and this frame rate is present in the metadata. 
If we have PTS and no other information we assume that this file might be variable frame rate and we need to have PTS of every frame to be able to decode it and show it correctly.